### PR TITLE
Feat/improve copy to clipboard

### PR DESCRIPTION
**Pull request by Jarsop**
Adding a new field (`copy_cmd`) in the default configuration and in the configuration file.
It allows to easily add a new copy command.

### CODE REVIEW

File: src/copy.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+/// Returns the default `CopyCmd` according to the conditional compilation on the detected OS
+/// which is the first item of the `Vec<CopyCmd>` obtained by `get_supported_copy_cmds()`.
+pub(crate) fn get_default_copy_cmd() -> CopyCmd {
+    let copy_cmds = get_supported_copy_cmds();
+    // We always have at least one element in this array
+    copy_cmds[0].to_owned()
+}
+
+pub(crate) fn deserialize_copy_cmd<'de, D>(deserializer: D) -> Result<CopyCmd, D::Error>
```
Comment (Ninjani) [inline src/copy.rs:50]: I would expect the value in the configuration file to contain the entire command with arguments (e.g "xclip -in -selection clipboard"). This way people can add in their own commands.

File: src/copy.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    pub(crate) cmd: &'static str,
+    /// Arguments to pass to the copy command
+    pub(crate) args: Vec<&'static str>,
+}
+
+/// Defines the supported clipboard copy command.
+/// A `Vec<CopyCmd>` is returned according to the conditional compilation on the detected OS.
+/// The first item declared is considered the default copy command.
+fn get_supported_copy_cmds() -> Vec<CopyCmd> {
+    #[cfg(target_os = "linux")]
```
Comment (Ninjani) [inline src/copy.rs:18]: Would be good to add a case where it doesn't match any of these OSes and returns an empty list

File: src/copy.rs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    }];
+    cmds
+}
+
+/// Returns the default `CopyCmd` according to the conditional compilation on the detected OS
+/// which is the first item of the `Vec<CopyCmd>` obtained by `get_supported_copy_cmds()`.
+pub(crate) fn get_default_copy_cmd() -> CopyCmd {
+    let copy_cmds = get_supported_copy_cmds();
+    // We always have at least one element in this array
+    copy_cmds[0].to_owned()
```
Comment (Ninjani) [inline src/copy.rs:47]: Instead, could make this return a Result and give a message like "No default command found for detected OS. Please add a supported command to your configuration file (as copy_cmd)"

File: tests/cli.rs [hunk 4]
```diff
@@ -476,7 +476,10 @@ fn copy_shell_script() -> color_eyre::Result<()> {
     Ok(())
 }
 
-fn copy_shell_script_rexpect(config_file: PathBuf) -> rexpect::errors::Result<()> {
+#[test]
+fn copy_shell_script_rexpect() -> rexpect::errors::Result<()> {
```
Comment (Ninjani) [inline tests/cli.rs:480]: The copy tests are the one called `copy` and `copy_shell_script` just above this function. Please revert this change as they call out to this function to test the interactive behavior.

**Review by Ninjani — CHANGES_REQUESTED** [review]
Thanks! I've added some comments

**Comment by Jarsop** [comment]
Hello @Ninjani, does it means that you want to support only one command by OS and just override the default command by the configuration field ?

**Comment by Ninjani** [comment]
Yes, that's what I had in mind, that way it's easy to swap out with xsel or wl-copy or anything the user wants. What do you think?

**Comment by Jarsop** [comment]
Sounds good too, maybe a bit harder to handling the errors properly but more generic. I think an example in the `README` for that case will be welcome :) (I will add it)

**Comment by Jarsop** [comment]
I think that I will also add an error kind for the case of we are on a non supported OS by default copy command

**Comment by Jarsop** [comment]
I will update the `README` soon, after your review

**Comment by Jarsop** [comment]
@Ninjani Any update about this topic?

File: src/errors.rs [hunk 5]
```diff
@@ -30,6 +30,13 @@ pub enum LostTheWay {
     SyntaxError { syntax: String },
     #[error("ClipboardError: Couldn't copy to clipboard - {message}")]
     ClipboardError { message: String },
+    #[error(
+        "NoDefaultCopyCommand: No default command found for detected OS. \
+        Please add a supported command to your configuration file (as copy_cmd)"
+    )]
+    NoDefaultCopyCommand,
+    #[error("EmptyCopyCommand: The field `copy_cmd` from configuration is empty")]
```
[the remark this answers is not in the document]
Comment (Jarsop) [reply src/errors.rs:38]: Yes, I agree! I've changed the way (no joke :smiley: ) to handle this.

**Comment by Jarsop** [comment]
> Sorry got caught up with work. Looks pretty good, just a minor change and we're good to go I think.

No problem we are all busy :)

File: src/errors.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Jarsop) [reply src/errors.rs]: But just to clarify the `EmptyCopyCommand` was only raised when the field is empty in the configuration file and not if the field is not present. But it's fixed.

**Review by Ninjani — APPROVED** [review]

**Comment by Ninjani** [comment]
Looks good! Would you like to update the README and CHANGELOG?

**Comment by Jarsop** [comment]
> Looks good! Would you like to update the README and CHANGELOG?

Yes I will do it ?

**Comment by Jarsop** [comment]
Done. I also added an example with `xsel` and I incremented the minor version (I don't know if you consider it a patch or a minor increment)

**Comment by Ninjani** [comment]
Thanks @Jarsop I'll make a new release this weekend

**Pull request merged**